observability/metrics/src/main/scala/com/daml/metrics/InstrumentedExecutorServiceMetrics.scala

```diff
@@ -0,0 +1,215 @@
+
+  @MetricDoc.Tag(
+    summary = "The number of tasks running in an instrumented executor.",
+    description = "Currently running number of tasks.",
+    qualification = Debug,
+    labelsWithDescription = Map(
+      "name" -> "The name of the executor service.",
+      "type" -> "The type of the executor service. Can be `fork_join` or `thread_pool`.",
+    ),
+  )
+  val running: Counter = factory.counter(
+    InstrumentedExecutorServiceMetrics.Prefix :+ "running",
+    "The number of currently running tasks.",
+  )
+
+  @MetricDoc.Tag(
+    summary = "The number of tasks completed in an instrumented executor.",
+    description = "Number of tasks that were completed by the executor.",
+    qualification = Debug,
+    labelsWithDescription = Map(
+      "name" -> "The name of the executor service.",
+      "type" -> "The type of the executor service. Can be `fork_join` or `thread_pool`.",
+    ),
+  )
+  val completed: Meter = factory.meter(
+    InstrumentedExecutorServiceMetrics.Prefix :+ "completed",
+    "Number of tasks completed by this executor",
+  )
+
+  @MetricDoc.Tag(
+    summary = "The time that a task is idle in an instrumented executor.",
+    description =
+      "A task is considered idle if it was submitted to the executor but it has not started execution yet.",
+    qualification = Debug,
+    labelsWithDescription = Map(
+      "name" -> "The name of the executor service.",
+      "type" -> "The type of the executor service. Can be `fork_join` or `thread_pool`.",
+    ),
+  )
+  val idle: Timer = factory.timer(
+    InstrumentedExecutorServiceMetrics.Prefix :+ "idle",
+    "Time passed since the tasks was submitted to the executor until it started execution.",
+  )
+
+  @MetricDoc.Tag(
+    summary = "The duration of a task is running in an instrumented executor.",
+    description = "A task is considered running only after it has started execution.",
+    qualification = Debug,
+    labelsWithDescription = Map(
+      "name" -> "The name of the executor service.",
+      "type" -> "The type of the executor service. Can be `fork_join` or `thread_pool`.",
+    ),
+  )
+  val duration: Timer = factory.timer(
+    InstrumentedExecutorServiceMetrics.Prefix :+ "duration",
+    "Time passed from the moment the task started execution until it has finished execution. Idle time is not considered as part of this timer.",
+  )
+}
+object InstrumentedExecutorServiceMetrics {
+
+  private val Prefix = ExecutorServiceMetrics.Prefix :+ "runtime"
+
+  /** Provides instrumentation for all the submissions to the executor service.
+    * Note:
+    * - We might want to change in the future how instrumentation for the `invokeAll`/`invokeAny` is done.
+    *    Currently we treat all the tasks as individual tasks but from an executor point of view
+    *    we might want to monitor the different semantics that those methods have (eg: 1 submission but multiple tasks,
+    *    timing when the slowest finishes or when the fastest finishes..)
```

Comment:
Thanks for adding this note here, @nicu-da. As this is more of a suggestion than describing how the current code works, I'd consider moving this into either the PRs description as potential follow-up or creating a ticket to track it.
Hence, you might want to reword the comment to focus on the current behavior, e.g.:
```suggestion
    * Note that when instrumenting the `invokeAll`/`invokeAny` methods we
    * currently treat all tasks as individual tasks and don't necessarily report a metric that makes sense semantically
    * (e.g., in case of 1 transaction made up of multiple tasks).
```